ansible/roles/nginx/templates/nginx.conf.j2

```diff
@@ -16,6 +16,15 @@ http {
         '$http_referer $http_user_agent $upstream_addr';
     access_log /logs/nginx_access.log combined-upstream;
 
+    upstream controllers {
+        server {{ groups['controllers'] | first }}:{{ controller.port }};
+{% for ip in groups['controllers'] %}
+    {% if groups['controllers'].index(ip) > 0 %}
+        server {{ ip }}:{{ controller.port + groups['controllers'].index(ip) }} backup;
```

Comment:
Will you address cache invalidation between primary and backup instances in this pr?